titiler/api/endpoints/mosaic.py

```diff
@@ -0,0 +1,399 @@
+from titiler.ressources.enums import (
+    ImageMimeTypes,
+    ImageType,
+    MimeTypes,
+    PixelSelectionMethod,
+)
+from titiler.ressources.responses import ImgResponse, XMLResponse
+
+from fastapi import APIRouter, Depends, Path, Query
+
+from starlette.concurrency import run_in_threadpool
+from starlette.requests import Request
+from starlette.responses import Response
+from starlette.templating import Jinja2Templates
+
+router = APIRouter()
+templates = Jinja2Templates(directory="titiler/templates")
+
+
+def _read_point(asset: str, *args, **kwargs) -> List:
+    """Read pixel value at a point from an asset"""
+    with rasterio.open(asset) as src_dst:
+        return point(src_dst, *args, **kwargs)
+
+
+@router.post(
+    "", response_model=MosaicJSON, response_model_exclude_none=True,
+)
+def create_mosaicjson(body: CreateMosaicJSON):
+    """Create a MosaicJSON"""
+    mosaic = MosaicJSON.from_urls(
+        body.files,
+        minzoom=body.minzoom,
+        maxzoom=body.maxzoom,
+        max_threads=body.max_threads,
+    )
+    mosaic_path = CommonMosaicParams(body.url).mosaic_path
+    with MosaicBackend(mosaic_path, mosaic_def=mosaic) as mosaic:
+        try:
+            mosaic.write()
+        except NotImplementedError:
+            raise BadRequestError(
+                f"{mosaic.__class__.__name__} does not support write operations"
+            )
+        return mosaic.mosaic_def
+
+
+@router.get(
+    "",
+    response_model=MosaicJSON,
+    response_model_exclude_none=True,
+    responses={200: {"description": "Return MosaicJSON definition"}},
+)
+def read_mosaicjson(resp: Response, mosaic_params: CommonMosaicParams = Depends()):
+    """Read a MosaicJSON"""
+    resp.headers["Cache-Control"] = "max-age-3600"
+    with MosaicBackend(mosaic_params.mosaic_path) as mosaic:
+        return mosaic.mosaic_def
+
+
+@router.put("", response_model=MosaicJSON, response_model_exclude_none=True)
+def update_mosaicjson(body: UpdateMosaicJSON):
+    """Update an existing MosaicJSON"""
+    mosaic_path = CommonMosaicParams(body.url).mosaic_path
+    with MosaicBackend(mosaic_path) as mosaic:
+        features = get_footprints(body.files, max_threads=body.max_threads)
+        try:
+            mosaic.update(features, add_first=body.add_first, quiet=True)
+        except NotImplementedError:
+            raise BadRequestError(
+                f"{mosaic.__class__.__name__} does not support update operations"
+            )
+        return mosaic.mosaic_def
+
+
+@router.get(
+    "/bounds",
+    response_model=cogBounds,
+    responses={200: {"description": "Return the bounds of the MosaicJSON"}},
+)
+def mosaicjson_bounds(resp: Response, mosaic_params: CommonMosaicParams = Depends()):
+    """Read MosaicJSON bounds"""
+    resp.headers["Cache-Control"] = "max-age=3600"
+    with MosaicBackend(mosaic_params.mosaic_path) as mosaic:
+        return {"bounds": mosaic.mosaic_def.bounds}
+
+
+@router.get("/info", response_model=cogInfo)
```

Comment:
It's a bit weird to use cogInfo for a mosaic metadata ?

Reply:
https://github.com/developmentseed/titiler/pull/40/commits/4602df3f26d527c3a5778a093f25d27e339040ff

Removed addition of cogInfo from mosaic metadata.  I think it is worth it to include some asset-level metadata about the mosaic like data type and photometric interpretation to name a few but this may be easier to implement after code dedup (ref #33).

Also this assumes that all assets in the MosaicJSON are the same, and I'm not sure if that assumption is always valid.

Reply:
yeah we can remove the `COG info` for now, IMO most of this should come from the mosaicJSON document ;-)